src/main/java/eu/europa/ec/fisheries/uvms/exchange/model/mapper/ExchangePluginRequestMapper.java

```diff
@@ -134,4 +134,18 @@ public static String createSendFLUXFAReportRequest(String request, String destin
         faqReq.setResponseLogGuid(responseMsgGuid);
         return JAXBMarshaller.marshallJaxBObjectToString(faqReq);
     }
+
+    public static String createSendFLUXMovementReportRequest(String report, String destination, String senderOrReceiver, String dataflow, String unsentMessageGuid) throws ExchangeModelMarshallException {
+        SendFLUXMovementReportRequest request = new SendFLUXMovementReportRequest();
+        request.setMethod(ExchangePluginMethod.SEND_MOVEMENT_REPORT);
+        request.setRecipient(request.getSenderOrReceiver());
+        request.setReport(report);
+        if (request.getSenderOrReceiver() != null) {
```

Comment:
Miss check? You wanted  if (senderOrReceiver != null) ? since request.senderOrReceiver is always null at that point

Reply:
Good observation, correcting!

Reply:
nice catch! it was meant to be set from the method parameters